### PR TITLE
US27 29

### PR DESCRIPTION
**Pull request by perryr16**
27: user profile has a link to display order (if order exists)
28: Orders index page has info about all orders associated with user
29: Orders show page has info about individual order
added column `status` to `item_orders` table. this is for `fulfilling` an order

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (perryr16) [inline .vscode/settings.json:1]: whoops, didnt know this got pushed up. I'll add it to .gitignore

File: app/models/item_order.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@ class ItemOrder <ApplicationRecord
   belongs_to :item
   belongs_to :order
 
+  enum status: {unfulfilled: 0, fulfilled: 1}
```
Comment (perryr16) [inline app/models/item_order.rb:7]: merchants can fulfill individual items on an order. Figured we should store it as an integer then take advantage of the enum method

Comment (foleymichelle9) [reply app/models/item_order.rb:7]: Nice!

**Comment by foleymichelle9** [comment]
I approve!

File: config/routes.rb [hunk 3]
```diff
@@ -42,4 +43,10 @@
   resources :sessions, only: [:create]
   get '/login', to: 'sessions#new'
   get '/logout', to: 'sessions#destroy'
+
+  resources :profile, only: [:show] do
```
Comment (perryr16) [inline config/routes.rb:47]: not all paths can be done with resources :(

File: db/migrate/20200528141507_add_status_to_orders.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :orders, :status, :integer, default: 0
```
Comment (perryr16) [inline db/migrate/20200528141507_add_status_to_orders.rb:3]: status of order is `pending` = 0, `packaged`, `shipped` and `cancelled`

File: db/migrate/20200528185110_add_status_to_item_orders.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToItemOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :item_orders, :status, :integer, default: 0
```
Comment (perryr16) [inline db/migrate/20200528185110_add_status_to_item_orders.rb:3]: merchants fulfill items one at a time on an order. that sounded like a ItemOrder column to me

File: spec/features/orders/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+RSpec.describe 'User Login-Logout' do
+  before :each do
+    @regular2 = create(:user, email: 'regular2@email.com')
+    @password = 'p123'
+    visit '/login'
+    within("#login-form")do
+      fill_in :email, with: @regular2.email
+      fill_in :password, with: @password
+      click_button "Login"
```
Comment (perryr16) [inline spec/features/orders/index_spec.rb:11]: how do I create a session in my spec test?

File: spec/features/orders/user_orders_show_spec.rb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  end
+
+  describe "US27" do
+    it "profile displays a MY ORDERS link if orders" do
+      visit '/profile'
+
+      expect(page).to_not have_link("My Orders")
+
+      visit "/items/#{@paper.id}"
+      click_on "Add To Cart"
```
Comment (perryr16) [inline spec/features/orders/user_orders_show_spec.rb:28]: could have just created item_orders. but hey, I was young when I wrote this

File: db/migrate/20200528141507_add_status_to_orders.rb (hunk 4 above)
Comment (dborski) [reply db/migrate/20200528141507_add_status_to_orders.rb:3]: Great idea using an enum for our order status too

File: spec/features/orders/index_spec.rb (hunk 6 above)
Comment (perryr16) [reply spec/features/orders/index_spec.rb:11]: cory said you cant!

**Pull request merged**